### PR TITLE
HWKAGENT-125 Support Enable/Disable/Restart Deployment commands

### PR DESCRIPTION
**Pull request by jshaughn**
- add new commands
- add serverGroup handling to DeploymentJbossASClient for enable, disable
- add restart handling to DeploymentJbossASClient
- add new server operations in resource type config files
  - NOTE, leaving deployment commands until MIQ is updated
    to use the new server-level commands. See hwkagent-126.

### CODE REVIEW

**Comment by jshaughn** [comment]
@jmazzitelli please review

File: hawkular-dmr-client/src/main/java/org/hawkular/dmrclient/DeploymentJBossASClient.java [hunk 1]
```diff
@@ -38,22 +38,27 @@ public DeploymentJBossASClient(ModelControllerClient client) {
...
 
-    private void enableDisableDeployment(String name, boolean enable) throws Exception {
+    private void enableDisableDeployment(String name, boolean enable, Set<String> serverGroups) throws Exception {
         DeploymentManager dm = DeploymentManager.Factory.create(getModelControllerClient());
         DeploymentResult result;
 
         if (enable) {
-            result = dm.deployToRuntime(SimpleDeploymentDescription.of(name));
+            SimpleDeploymentDescription sdd = SimpleDeploymentDescription.of(name);
+            sdd.addServerGroups(serverGroups);
```
Comment (jmazzitelli) [inline hawkular-dmr-client/src/main/java/org/hawkular/dmrclient/DeploymentJBossASClient.java:58]: what if serverGroups is null? Does the API accept that? Or might be best to do:

sdd.addServerGroups(serverGroups != null ? serverGroups : Collections.emptySet());

File: hawkular-dmr-client/src/main/java/org/hawkular/dmrclient/DeploymentJBossASClient.java [hunk 2]
```diff
@@ -38,22 +38,27 @@ public DeploymentJBossASClient(ModelControllerClient client) {
...
+            result = dm.deployToRuntime(sdd);
         } else {
-            result = dm.undeploy(UndeployDescription.of(name).setRemoveContent(false));
+            UndeployDescription ud = UndeployDescription.of(name);
+            ud.addServerGroups(serverGroups);
```
Comment (jmazzitelli) [inline hawkular-dmr-client/src/main/java/org/hawkular/dmrclient/DeploymentJBossASClient.java:62]: what if serverGroups is null? Does the API accept that? Or might be best to do:

sdd.addServerGroups(serverGroups != null ? serverGroups : Collections.emptySet());

File: hawkular-dmr-client/src/main/java/org/hawkular/dmrclient/DeploymentJBossASClient.java [hunk 3]
```diff
@@ -63,6 +68,21 @@ private void enableDisableDeployment(String name, boolean enable) throws Excepti
         return; // everything is OK
     }
 
+    public void restartDeployment(String name, Set<String> serverGroups) throws Exception {
+        DeploymentManager dm = DeploymentManager.Factory.create(getModelControllerClient());
+        DeploymentResult result;
+
+        SimpleDeploymentDescription sdd = SimpleDeploymentDescription.of(name);
+        sdd.addServerGroups(serverGroups);
```
Comment (jmazzitelli) [inline hawkular-dmr-client/src/main/java/org/hawkular/dmrclient/DeploymentJBossASClient.java:82]: what if serverGroups is null? Does the API accept that? Or might be best to do:

sdd.addServerGroups(serverGroups != null ? serverGroups : Collections.emptySet());

**Comment by jmazzitelli** [comment]
other than my NPE checks, LGTM

**Pull request merged**